impl/src/settings.rs

```diff
@@ -483,77 +468,299 @@ fn validate_settings(
+      || self.genmode.is_some()
+      || self.output_buildfile_suffix.is_some()
+      || self.default_gen_buildrs.is_some()
+      || self.registry.is_some()
+      || self.index_url.is_some()
+      || self.incompatible_relative_workspace_path.is_some()
+      || self.rust_rules_workspace_name.is_some()
+      || self.vendor_dir.is_some()
+  }
+}
+
+/** Grows a list with duplicate keys between two maps */
+fn extend_duplicates<K: Hash + Eq + Clone, V>(
+  extended_list: &mut Vec<K>,
+  main_map: &HashMap<K, V>,
+  input_map: &HashMap<K, V>,
+) {
+  extended_list.extend(
+    input_map
+      .iter()
+      .filter_map(|(key, _value)| {
+        // Log the key if it exists in both the main and input maps
+        if main_map.contains_key(key) {
+          Some(key)
+        } else {
+          None
+        }
+      })
+      .cloned()
+      .collect::<Vec<K>>(),
+  );
+}
+
+/** Parse [`RazeSettings`](crate::settings::RazeSettings) from workspace metadata */
+fn parse_raze_settings_workspace(
+  metadata_value: &serde_json::value::Value,
+  metadata: &Metadata,
+) -> Result<RazeSettings> {
+  let mut settings = RazeSettings::deserialize(metadata_value)?;
+
+  let workspace_packages: Vec<&Package> = metadata
+    .packages
+    .iter()
+    .filter(|pkg| metadata.workspace_members.contains(&pkg.id))
+    .collect();
+
+  let mut duplicate_binary_deps = Vec::new();
+  let mut duplicate_crate_settings = Vec::new();
+
+  for package in workspace_packages.iter() {
+    if let Some(pkg_value) = package.metadata.get("raze") {
+      let pkg_settings = RawRazeSettings::deserialize(pkg_value)?;
+      if pkg_settings.contains_primary_options() {
+        return Err(anyhow!(
+          "The package '{}' contains Primary raze settings, please move these to the \
+           `[workspace.metadata.raze]`",
+          package.name
+        ));
+      }
+
+      // Log duplicate binary dependencies
+      extend_duplicates(
+        &mut duplicate_binary_deps,
+        &settings.binary_deps,
+        &pkg_settings.binary_deps,
       );
+
+      settings
+        .binary_deps
+        .extend(pkg_settings.binary_deps.into_iter());
+
+      // Log duplicate crate settings
+      extend_duplicates(
+        &mut duplicate_crate_settings,
+        &settings.crates,
+        &pkg_settings.crates,
+      );
+
+      settings.crates.extend(pkg_settings.crates.into_iter());
     }
-    return Ok(raze);
   }
 
-  // Next is package
-  if let Some(raze) = toml_contents
-    .package
-    .and_then(|pkg| pkg.metadata.and_then(|data| data.raze))
-  {
-    if toml_contents.raze.is_some() {
-      eprintln!(
-        "WARNING: Both [raze] and [package.metadata.raze] are set. Using [package.metadata.raze] \
-         and ignoring [raze], which is deprecated."
-      );
+  // Check for duplication errors
+  if duplicate_binary_deps.len() > 0 {
+    return Err(anyhow!(
+      "Duplicate `raze.binary_deps` values detected accross various crates: {:?}",
```

Comment:
Hmm..

I think we should permit duplicate binary deps provided they match. This serves as a record for humans as well, and they might be confused having to decide which toml to put the binary dependency in.

(i dont feel strongly about this)

Reply:
I think if something is used by multiple packages, then it should go in workspaces' metadata. I think allowing users to specify it in multiple packages promotes some sloppy copy/pasting of configurations. I'd like for this to be well defined. I'd opt to leave it as is but if you feel stronger about this I can make the change since the benefit of your suggestion is that users can see more relevant information about their package.

PS: Ultimately, my hope is that the need for this functionality will some day go away (if Cargo ever figures out what it wants to do about these things).

Reply:
I think the right answer here depends on the expected mental model. If users associate the binary dependency with an individual sub-crate, then it might make more sense for this setting permit duplicates, since you might have multiple crates using the given binary dependency. If they view it as boilerplate for raze which is mostly independent of the subcrate in question then it might make more sense to prohibit duplicates (though, in that case I'd argue it ought to be *required* to be in the workspace root toml).

I still don't feel strongly about this though, so final decision up to you (though I'd recommend trying to find a place to document this restriction in the README if it isn't already documented).

Reply:
I've updated the documentation. I do think you're instincts are toward the better overall outcome here, but I would rather keep the functionality of binary dependencies very minimal and find a way to remove it. Maybe I'll try to drive some RFCs forward in cargo to try and get this resolved. 

Anyway, I think it's a much smaller change to make interactions here more elegant and I'd love to find out that people feel strongly enough about it to make a PR for it 😄